### DISCUSSION TITLE
Material property has same name as an existing variable error

### DISCUSSION THREAD

**Discussion by AndrewFalkowski** [idaholab/moose#24065] [Q&A Modules: Solid mechanics] [answered]
Hello,

I am running an input file that pulls from the tensor mechanics module and am getting the following error: `The material property output max_principal_stress has the same name as an existing variable, either use the material declare_suffix parameter to disambiguate or the output_properties parameter to restrict the material properties to output`. I haven't run into this issue before with similar input files, and I can't seem to find where this variable is being previously declared or what material objects are producing a variable with a similar name. Any advice on how to get around this? 

I've included relevant code below for a problem wherein a material has swelled in response to the diffusion of a species into it.:

```
[GlobalParams]
  displacements = 'disp_r'
[]
[Mesh]
  coord_type = RZ
  [gen]
    type = FileMeshGenerator
    file = 'outputs/Diffusion_Profile.e'
    use_for_exodus_restart = true
    show_info = false
  []
[]

[AuxVariables]
  [C]
    initial_from_file_var = C
  []
[]

[Kernels]
[]

[AuxKernels]
[]

[Modules/TensorMechanics/Master]
  [all]
    strain = SMALL
    add_variables = true
    use_automatic_differentiation = false
    automatic_eigenstrain_names = true
    material_output_order = SECOND
    generate_output = 'vonmises_stress max_principal_stress min_principal_stress'
  []
[]


[BCs]
  [no_disp]
    type = DirichletBC
    variable = disp_r
    boundary = left
    value = 0.0
  []
[]

[Materials]
  [mat_E]
    type = PiecewiseLinearInterpolationMaterial
    variable = C
    property = E_mat
    x = '0 2'
    y = '95 130'
    scale_factor = 1e9
  []
  [mat_elastic_properties]
    type = ComputeVariableIsotropicElasticityTensor
    args = C
    poissons_ratio = 0.32
    youngs_modulus = E_mat
  []
  [swelling] # using thermal expansion as analogue 
    type = ComputeThermalExpansionEigenstrain
    eigenstrain_name = thermal_expansion
    stress_free_temperature = 0
    temperature = C
    thermal_expansion_coeff = 0.07
  []
  [stress]
    type = ComputeLinearElasticStress
  []
[]
[Executioner]
  type = Steady
  solve_type = PJFNK
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
  automatic_scaling = true
  compute_scaling_once = false
[]

[Outputs]
  [exo]
    type = Exodus
    output_material_properties = true
    file_base = 'user/outputs/Stress_Profile'
  []
[]
```

**Comment by GiudGiud** [comment]
Hello

It s coming from this line
`    generate_output = 'vonmises_stress max_principal_stress min_principal_stress'`
you could just remove that one from the output

Guillaume

**Reply by AndrewFalkowski** [reply]
Thank you for the quick response. I am a little confused though as I have never encountered this issue in similar input files where I include that line. I don't see why that would be causing an issue. Additionally, when I remove, I get a new error: 

```
*** ERROR ***
The following error occurred in the object "all_strain", of type "ComputeAxisymmetricRZSmallStrain".

For 2D simulations where the out-of-plane direction is z the number of supplied displacements must be two.
```

This is a 1D simulation, so it should only need one displacement variable. Not sure what could be causing this.

**Reply by GiudGiud** [reply]
RZ is 2D though. Is your mesh 1D?
You could supply a second dummy displacement for now.

**Reply by AndrewFalkowski** [reply]
My understanding was that RZ was simply specifying the coordinate type (in this case, cylindrical), not the number of dimensions in the simulation.

How does one supply a dummy displacement? Adding a dummy `disp_z` through `[GlobalParams]` throws an error because the mesh is only 1D.

**Reply by GiudGiud** [reply]
Oh for sure, the dimension is set by the mesh.
It seems that this material / strain model is expecting 2D though.

Looks like you cant use a dummy displacement unless you make your mesh trivial in an additional dimension.

**Reply by AndrewFalkowski** [reply]
Hmm ok, I would think it would be possible to compute that quantity over a 1D model.

> It s coming from this line
    generate_output = 'vonmises_stress max_principal_stress min_principal_stress'
you could just remove that one from the output

Returning to my original inquiry, removing this line also removes these calculated values from my output. While I can certainly calculate these quantities from the tensor outputs, I shouldn't have to and I haven't had to remove this line from Tensor Mechanics sims I have done in the past. Is there a material object that might be overwriting or predefining them somewhere?

**Reply by GiudGiud** [reply, marked as answer]
so you have both
    output_material_properties = true
and 
    generate_output = 'vonmises_stress max_principal_stress min_principal_stress'
could those be conflicting? Can you try only the latter

**Reply by GiudGiud** [reply]
otherwise seems like a bug and we d open an issue

**Reply by AndrewFalkowski** [reply]
That seems to be the problem, thank you!
